### ISSUE TITLE
Updates to plot do not cascade to colorlegend 

### ISSUE THREAD

**Issue by yakir12** [JuliaPlots/AbstractPlotting.jl#198]
This is from a conversation on [slack#makie](https://julialang.slack.com/archives/C8RQUU2KH/p1567521043076600).

Updating the colors of a plot does not update the properties of the plot's color-legend (colorbar). 

More specifically, if I `colorlegend!` a `surface` plot who's colors are `lift`ed from a `node`, when updating the node, the colors in the surface change, but the colrobar's limits are wrong. So it seems like the color limits for the colorbar are determined at plotting time, and then they don't get updated after the plot they are connected to does.

In this silly MWE, the colors shouldn't ever change (cause the color limits are normalized), but the top limit on the colorbar should. It only changes if you rerun this with different starting values for the slider:

```julia
sc = Scene()
h, sl = textslider(1:10, "text", 
    start = 2 # change this to get different values in the colorbar's maximum 
) 
_x = collect(1:10)
x = _x .+ 0*_x'
y = 0*_x .+ _x'
z = rand(10, 10)
p = surface!(sc, x, y, z, color = lift(x -> repeat(range(0, x, length = 10), outer = [1,10]), sl))
cb = colorlegend(p[end], width = (30, 300))
vbox(hbox(h, p), cb)
```

**Comment by asinghvi17** [comment]
https://github.com/JuliaPlots/AbstractPlotting.jl/blob/2e88c9d4437556c39e266cb97eac2fabd2623834/src/basic_recipes/legend.jl#L178

**Comment by SimonDanisch** [comment]
Nope ;) The easy hack would be , to not define this conversion, and check in the body of the recipe what kind of arguments we got... but that's not as elegant of course ;)

**Comment by asinghvi17** [comment]
Yep, but there doesn't seem to be another solution (save for providing an accessor function), so 🤷‍♂

**Comment by asinghvi17** [comment]
Wasn't there a way to specialize a recipe on the input types?
